### DISCUSSION TITLE
How do I get the deepspeed env file not to be re-generated?

### DISCUSSION THREAD

**Discussion by shing100** [axolotl-ai-cloud/axolotl#1948] [Q&A]
There are a lot of nodes on the host, so when running as a multi-node, the deep speed env file continues to be created, causing problems.
Argument list too long: 'pdsh'

<img width="725" alt="스크린샷 2024-10-07 오후 7 32 52" src="https://github.com/user-attachments/assets/ab042778-85ef-4fa6-afd3-afcf179fd9eb">

Is there a solution?

**Comment by NanoCode012** [comment]
Hello, could you clarify what you mean by deepspeed env file? Is it the `zero*.json`?

**Comment by shing100** [comment]
<img width="164" alt="스크린샷 2024-10-09 오후 1 39 54" src="https://github.com/user-attachments/assets/497b2c21-5601-4339-abe2-bc766d48eee8">

this file '.deepspeed_env'

**Reply by NanoCode012** [reply]
Thanks for clarifying. The issue is that your cli command being passed to pdsh is too long, am I understanding it right? 

I'm not so familiar with this issue. Could you give me an example of how you run the command?

**Reply by shing100** [reply]
yes!

multi-node setting and run training (accelerate launch -m axolotl.cli.train your_config.yml)

**Reply by NanoCode012** [reply]
@shing100 , could you perhaps open a discussion on Deepspeed's side to see what their recommendation is? I don't think this issue is from axolotl, and they may have more expertise on this.

**Reply by shing100** [reply]
okay thanks!!

**Reply by NanoCode012** [reply]
I'm closing this for now but do update us if you do find a workaround :)
